src/runtime/node/process/_process.ts

```diff
@@ -262,3 +266,125 @@ process.hrtime.bigint = function () {
   // Convert milliseconds to nanoseconds
   return BigInt(Date.now() * 1_000_000);
 };
+
+let egid = 0;
+process.getegid = function () {
+  return egid;
+};
+process.setegid = function (val: number) {
+  egid = val;
+};
+
+let euid = 0;
+process.geteuid = function () {
+  return euid;
+};
+process.seteuid = function (val: number) {
+  euid = val;
+};
+
+let gid = 0;
+process.getgid = function () {
```

Comment:
We could default to a non zero value (like `1000`) for instead of `0` (root). In case of any dependency of this to do an operation I think it might be safer..